enos/provider/g5k.py

```diff
@@ -218,56 +273,88 @@ def _check_nodes(self,
             wanted_nodes += reduce(operator.add, map(int, roles.values()))
 
         if mode == ROLE_DISTRIBUTION_MODE_STRICT and wanted_nodes > len(nodes):
-            raise Exception("Not enough nodes to continue")
+            raise Exception("Not enough nodes to continue (%d missing)"
+                    % (wanted_nodes - len(nodes)))
 
         return True
 
-    def _get_job(self):
-        """Get the hosts from an existing job (if any) or from a new job.
-        This will perform a reservation if necessary."""
+    def _get_job_ids(self):
```

Comment:
+1 for purely functional util functions. Referential transparency helps reasoning and will help modularization if we decide to move such util functions in the util package.